app/browser/tabs.js

```diff
@@ -1019,9 +1021,34 @@ const api = {
           createProperties.parent_partition = ''
         }
         if (createProperties.isTor) {
+          // TODO(riastradh): Duplicate logic in app/filtering.js.
           createProperties.isolated_storage = true
           createProperties.parent_partition = ''
-          createProperties.tor_proxy = 'socks5://127.0.0.1:9050'
+          let portno
+          switch (channel()) {
+            case 'dev':
+            case '':
+            default:
+              portno = 9250
+              break
+            case 'beta':
+              portno = 9260
+              break
+            case 'nightly':
+              portno = 9270
+              break
+            case 'developer':
+              portno = 9280
+              break
+          }
+          createProperties.tor_proxy = `socks5://127.0.0.1:${portno}`
+          const userDataDir = app.getPath('userData')
+          try {
+            fs.mkdirSync(path.join(userDataDir, 'tor'), 0o0700)
```

Comment:
you should make this directory on startup if it doesn't exist, instead of every time that a tor tab is created, for performance reasons. see app/sessionStore.js